src/main/java/io/github/pylonmc/pylon/base/content/machines/smelting/SmelteryHopper.java

```diff
@@ -31,6 +35,15 @@ public void onItemMoveFrom(InventoryMoveItemEvent event) {
         event.setCancelled(true);
     }
 
+    @Override
+    public void onBreak(@NotNull List<@NotNull ItemStack> drops, @NotNull BlockBreakContext context) {
+        Hopper hopper = (Hopper) getBlock().getState();
+
+        for (ItemStack item : hopper.getInventory()) {
+            drops.add(item);
+        }
```

Comment:
```suggestion
        for (ItemStack item : hopper.getInventory()) {
            if (item != null) {
                drops.add(item);
            }
        }
```

Item can't be null exception when broken with empty slots